path_trie.go

```diff
@@ -14,10 +14,29 @@ type PathTrie struct {
 	children  map[string]*PathTrie
 }
 
+// PathTrieConfig for building a path trie with different segmenter
+type PathTrieConfig struct {
+	segmenter StringSegmenter
```

Comment:
Your original aim was to not have this be an internal field

Reply:
oops, it's a typo.